src/rgw/rgw_sal_motr.cc

```diff
@@ -2134,11 +2132,16 @@ int MotrObject::read_mobj(const DoutPrefixProvider* dpp, int64_t off, int64_t en
 
   left = end - off;
   for (; left > 0; off += actual) {
-    if (left < bs)
-      bs = this->get_optimal_bs(left);
     actual = bs;
-    if (left < bs)
+    if (left < bs) {
+      uint64_t lid = M0_OBJ_LAYOUT_ID(meta.layout_id);
+      unsigned unit_sz = m0_obj_layout_id_to_unit_size(lid);
+      bs = roundup(left, unit_sz);
       actual = left;
+    }
+
+    mobj->ob_entity.en_flags |= M0_OOF_HOLE;
```

Comment:
This is wrong. This may end up with corrupted data. See more at https://github.com/Seagate/cortx-motr/pull/1728.

Reply:
@andriytk , this was conveyed to me by @yeshpal-jain-seagate that every read should be with this flag and that will help in reading the data when it was written in degraded mode. May be we need more clarity on this